evm_arithmetization/src/cpu/kernel/tests/core/access_lists.rs

```diff
@@ -10,82 +11,213 @@ use crate::cpu::kernel::constants::global_metadata::GlobalMetadata::{
     AccessedAddressesLen, AccessedStorageKeysLen,
 };
 use crate::cpu::kernel::interpreter::Interpreter;
-use crate::memory::segments::Segment::{AccessedAddresses, AccessedStorageKeys};
+use crate::memory::segments::Segment::{self, AccessedAddresses, AccessedStorageKeys};
+use crate::memory::segments::SEGMENT_SCALING_FACTOR;
 use crate::witness::memory::MemoryAddress;
 
+#[test]
+fn test_init_access_lists() -> Result<()> {
+    let init_label = KERNEL.global_labels["init_access_lists"];
+
+    // Check the initial state of the access list in the kernel.
+    let initial_stack = vec![0xdeadbeefu32.into()];
+    let mut interpreter = Interpreter::<F>::new_with_kernel(init_label, initial_stack);
+    interpreter.run()?;
+
+    assert!(interpreter.stack().is_empty());
+
+    let acc_addr_list: Vec<U256> = (0..2)
+        .map(|i| {
+            interpreter.generation_state.memory.get(MemoryAddress::new(
+                0,
+                Segment::AccessedAddresses,
+                i,
+            ))
+        })
+        .collect();
+    assert_eq!(
+        vec![U256::MAX, (Segment::AccessedAddresses as usize).into(),],
+        acc_addr_list
+    );
+
+    let acc_storage_keys: Vec<U256> = (0..4)
+        .map(|i| {
+            interpreter.generation_state.memory.get(MemoryAddress::new(
+                0,
+                Segment::AccessedStorageKeys,
+                i,
+            ))
+        })
+        .collect();
+
+    assert_eq!(
+        vec![
+            U256::MAX,
+            U256::zero(),
+            U256::zero(),
+            (Segment::AccessedStorageKeys as usize).into()
+        ],
+        acc_storage_keys
+    );
+
+    Ok(())
+}
+
+#[test]
+fn test_list_iterator() -> Result<()> {
+    let init_label = KERNEL.global_labels["init_access_lists"];
+
+    let initial_stack = vec![0xdeadbeefu32.into()];
+    let mut interpreter = Interpreter::<F>::new_with_kernel(init_label, initial_stack);
+    interpreter.run()?;
+
+    // test the list iterator
+    let mut list = interpreter
+        .generation_state
+        .get_addresses_access_list()
+        .expect("Couldn't retrieve access list");
```

Comment:
I remember @dvdplm mentioning that the argument in `expect` should mention why it _shouldn't_ panic. It's already not the case in other parts of the code, is it something we should strive to maintain?

Reply:
I'd say let's try to not increase the surface of code to address in the future, so any new `expect` / `unwrap` should follow the expected guidelines. Same as, say, new methods, that should always come with documentation, unless they are trivial one-liners.